table/block_based_table_reader.cc

```diff
@@ -1731,9 +1732,40 @@ bool BlockBasedTable::PrefixMayMatch(const Slice& internal_key,
   if (!prefix_extractor->InDomain(user_key)) {
     return true;
   }
-  assert(rep_->table_properties->prefix_extractor_name.compare(
-             prefix_extractor->Name()) == 0);
   auto prefix = prefix_extractor->Transform(user_key);
+  if (rep_->table_properties->prefix_extractor_name.compare(
+          prefix_extractor->Name()) != 0) {
```

Comment:
We should avoid this extra string comparator.

Reply:
I guess in `BlockBasedTable::NewIterator` we could store `prefix_extractor_changed` somewhere and use it here to avoid an extra string comparison